### PR TITLE
Enable file name and content disposition to be overridden

### PR DESCRIPTION
**Pull request by michalkleiner**
Other code can reuse the secure file controller and build on top of its functionality, e.g. providing a custom mangled file name or set the content disposition based on other attributes per each file download.

### CODE REVIEW

File: code/SecureFileController.php [hunk 1]
```diff
@@ -125,10 +125,16 @@ protected function disableFilters($callback) {
...
+	 *
 	 * For performance reasons, we avoid SS_HTTPResponse and just output the contents instead.
+	 *
+	 * @param $file                 File to output
+	 * @param string $name          Optional filename override value
+	 * @param string $disposition   Optional content disposition override value
+	 * @return bool|string|void
 	 */
-	public function sendFile($file) {
+	public function sendFile($file, $name = '', $disposition = '') {
```
Comment (dhensby) [inline code/SecureFileController.php]: can we use `null` as default values here, please

File: code/SecureFileController.php [hunk 2]
```diff
@@ -139,12 +145,16 @@ public function sendFile($file) {
 			return file_get_contents($path);
 		}
 
-		$disposition = $this->config()->content_disposition;
-		if(!$disposition) $disposition = 'attachment';
+		if(!$disposition) {
+			$disposition = $this->config()->content_disposition;
+			if (!$disposition) $disposition = 'attachment';
```
Comment (dhensby) [inline code/SecureFileController.php]: This breaks our coding style rules. A ternary operation would be more appropriate `$disposition = $this->config()->content_disposition ?: 'attachment';`

File: code/SecureFileController.php [hunk 3]
```diff
@@ -139,12 +145,16 @@ public function sendFile($file) {
...
+		}
+
+		if(!$name) $name = basename($path);
```
Comment (dhensby) [inline code/SecureFileController.php]: Curly braces needed

Comment (michalkleiner) [reply code/SecureFileController.php]: Well, I stayed within the coding standard of the file, the whole module could use an update. But I'll update it.

File: code/SecureFileController.php (hunk 2 above)
Comment (michalkleiner) [reply code/SecureFileController.php]: I didn't change style of code that I didn't add or remove, but I'll update it.

**Comment by dhensby** [comment]
thanks :)

**Comment by michalkleiner** [comment]
@dhensby, do you want me to squash it and force update or create a new PR so it's cleaner?

**Comment by dhensby** [comment]
squash and push please :)

**Pull request merged**